core/node/da_clients/src/avail.rs

```diff
@@ -0,0 +1,404 @@
+use std::fmt::Debug;
+
+use async_trait::async_trait;
+use jsonrpsee::{
+    core::client::{Client, ClientT, Subscription, SubscriptionClientT},
+    rpc_params,
+    ws_client::WsClientBuilder,
+};
+use parity_scale_codec::{Compact, Decode, Encode};
+use scale_encode::EncodeAsFields;
+use subxt_signer::{
+    bip39::Mnemonic,
+    sr25519::{Keypair, Signature},
+};
+use zksync_config::AvailConfig;
+use zksync_da_client::{
+    types::{DAError, DispatchResponse, InclusionData},
+    DataAvailabilityClient,
+};
+
+const PROTOCOL_VERSION: u8 = 4;
+
+/// An implementation of the `DataAvailabilityClient` trait that interacts with the Avail network.
+#[derive(Debug)]
+pub struct AvailClient {
+    config: AvailConfig,
+    keypair: Keypair,
+}
+
+// Utility type needed for encoding the call data
+#[derive(parity_scale_codec::Encode, scale_encode::EncodeAsType)]
+#[encode_as_type(crate_path = "scale_encode")]
+pub struct SubmitData {
```

Comment:
Do these types have to be public? The least possible visibility should be used by default, so if you can make it private or at least `pub(crate)`, it would be much preferred.

Reply:
done